src/controllers/Auth.js

```diff
@@ -169,6 +175,148 @@ class AuthController {
       });
     }
   }
+
+  /**
+   *
+   * @description User should recieve a reset token email
+   * @constructor
+   * @static
+   * @param {object} req
+   * @param {object} res
+   * @returns {object} res
+   * @memberof AuthController
+   */
+  static async sendResetToken(req, res) {
+    try {
+      const {
+        email
+      } = req.body;
+      const errors = {};
+
+      // CHECK IF EMAIL EXIST
+      const findUser = await User.findOne({
+        where: {
+          email
+        }
+      });
+
+      if (findUser) {
+        if (findUser.verificationToken === '') {
+          const token = req.generate;
+          await User.update({
+            verificationToken: token
+          }, {
+            where: {
+              email
+            }
+          });
+
+          const url = `${req.protocol}://${req.get('host')}/api/v1/auth/resetPassword?resetToken=${token}`;
+          const html = resetTemplate(findUser.userName, url);
+
+          if (!isTest) {
+            await sendEmail
+              .sendEmail(
+                'do_not_reply@authorhaven.com',
+                email,
+                'Password Reset',
+                html
+              );
+          }
+
+          // RETURN SUCCESS IF SUCCESSFUL
+          return res.status(200).json({
+            status: 200,
+            message: 'reset code successfully sent to email',
+          });
+        }
+
+        const url = `${req.protocol}://${req.get('host')}/api/v1/auth/resetPassword?resetToken=${findUser.verificationToken}`;
+        const html = resetTemplate(findUser.userName, url);
+
+        if (!isTest) {
+          await sendEmail
+            .sendEmail(
+              'do_not_reply@authorhaven.com',
+              email,
+              'Resend Password Reset',
+              html
+            );
+        }
+
+        // RETURN SUCCESS IF SUCCESS
+        return res.status(200).json({
+          status: 200,
+          message: 'reset code resent to your email',
+        });
+      }
+
+      // SET ERROR IF EMAIL DOES NOT EXIST
+      errors.email = 'email does not exist';
+      return res.status(404).json({
+        status: 404,
+        message: errors,
+      });
+    } catch (err) {
+      return res.status(500).json({
+        status: 500,
+        message: 'Internal server error',
+      });
+    }
+  }
+
+  /**
+   *
+   * @description User should be able to reset their password with the token
+   * @constructor
+   * @static
+   * @param {object} req
+   * @param {object} res
+   * @returns {object} res
+   * @memberof AuthController
+   */
+  static async resetPassword(req, res) {
+    try {
+      const {
+        resetToken
+      } = req.query;
+      const {
+        password
+      } = req.body;
+
+      // find the token if it exist
```

Comment:
What's the use of the middleware created to verify a token, if you're still going to check if the token is also here?

Reply:
Because I am expiring the token after 1hour... So if they don't use the token before then, it should expire and not let the user reset their password... They have to request a new token... So in the verify, I am verifying if it is still valid or expired